services/Donkey/src/donkey/clients/data_info.clj

```diff
@@ -87,12 +89,17 @@
         (get path))))
 
 (defn rename
+  "Uses the data-info set-name endpoint to rename a file within the same directory."
   [params body]
-  (let [url (url/url (cfg/data-info-base-url) "data" "rename")
-        req-map {:query-params (select-keys params [:user])
-                 :content-type :json
-                 :body         (json/encode body)}]
-    (http/post (str url) req-map)))
+  (with-jargon (icat/jargon-cfg) [cm]
+    (if (not= (ft/dirname (:dest body)) (ft/dirname (:source body)))
+      (throw+ {:error_code error/ERR_BAD_OR_MISSING_FIELD}))
```

Comment:
It would be a good idea to include a message in this error about why the request was bad.
Maybe use a `clojure-commons.assertions/assert-valid` call that can create the exception for you.